src/polygon.jl

```diff
@@ -0,0 +1,508 @@
+        while j >= xmin
+            
+            if complex(j, i) in triangle # Right point of triangle found
+
+                j_right = j
+                rightslope = j - lastright
+                lastright = j
+                j = xmin # Out of the loop
+                j_left = max(j_left, 1)
+                j_right = min(j_right, n)
+                img[i, j_left:j_right] .= triangle.color # Draw the line between the 2 points you found
+                # We do this in the if statement because we don't want to draw a line if no point of the triangle was found on this line
+
+            end
+            j -= 1
+        end
+
+    end
+
+    return img
+end
+
+"""
+    drawimage(triangles::Array, canvas::Array)
+
+Goes over all triangles in an array and draws them all on the same canvas, then returns the resulting image. 
+"""
+function drawimage(triangles::Array, canvas::Array)
+    polyimg = copy(canvas)
+    for triangle in triangles
+        polyimg = drawtriangle(triangle, polyimg)
+    end
+    return polyimg
+end
+
+"""
+    colordiffsum(img1::Array, img2::Array, m::Int, n::Int)
+
+The objective function of the algorithm. Computes the difference between 2 images by computing the difference in color for every pixel.
+"""
+colordiffsum(img1::Array, img2::Array) = sum(colordiff.(img1, img2))
+
+"""
+    checktriangle(triangle::Triangle, m::Int, n::Int)
+
+Checks a triangle to make sure we don't have any STUPID triangles.
+A stupid triangle is a triangle which:
+- Is much too thin (a couple of pixels thin)
+- Has all 3 points outside of the canvas
+- Is very stretched out
+These stupid triangles can all mess with the drawtriangle function, which is why checking for them and weeding them out is essential.
+Otherwise we get errors and boy do we hate errors.
+"""
+function checktriangle(triangle::Triangle, m::Int, n::Int)
+    # Rather than using the actual boundaries of the canvas, we use a slightly smaller rectangle for safety reasons
+    s = 10 # safety factor
+    x1, x2, x3 = real(triangle.p1), real(triangle.p2), real(triangle.p3)
+    y1, y2, y3 = imag(triangle.p1), imag(triangle.p2), imag(triangle.p3)
+
+    max_y = max(y1, y2, y3)
+    min_y = min(y1, y2, y3)
+    max_delta_y = max_y - min_y
+    YisStupid = max_delta_y <= s && return true # Too thin
+
+    max_x = max(x1, x2, x3)
+    min_x = min(x1, x2, x3)
+    max_delta_x = max_x - min_x
+    XisStupid = max_delta_x <= s && return true # Too thin
+
+    p1_out_of_canvas = (x1 <= s || x1 >= (n - s)) || (y1 <= s || y1 >= (m - s))
+    p2_out_of_canvas = (x2 <= s || x2 >= (n - s)) || (y2 <= s || y2 >= (m - s))
+    p3_out_of_canvas = (x3 <= s || x3 >= (n - s)) || (y3 <= s || y3 >= (m - s))
+    pointsAreStupid = all([p1_out_of_canvas, p2_out_of_canvas, p3_out_of_canvas]) && return true # All points are (almost) outside of the canvas, that's no bueno
```

Comment:
I think these code blocks can be extracted into dedicated functions. `checktriangle` has too many return statements now, it's a bit hard to follow the control flow.

Reply:
I think this is the only comment I didn't integrate for a good reason because I'm running out of time. Sorry 😕